plugins/events-correlation-engine/src/main/java/org/opensearch/plugin/correlation/core/index/serializer/CorrelationVectorSerializer.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.correlation.core.index.serializer;
+
+import org.opensearch.ExceptionsHelper;
+import org.opensearch.common.io.stream.BytesStreamInput;
+import org.opensearch.common.io.stream.BytesStreamOutput;
+
+import java.io.IOException;
+
+/**
+ * CorrelationVectorSerializer class to do serde operation of converting float vectors to byte array and vice versa.
+ *
+ * @opensearch.internal
+ */
+public class CorrelationVectorSerializer {
```

Comment:
Do you need a separate class to hold these functions? Why can't `floatToByteArray` be a function inside VectorField since that appears to be the only place it is used?

`byteToFloatArray` seems to only be used in tests, so can it be moved to a test utility?

Reply:
moved `floatToByteArray` to `VectorField` class & moved `byteToFloatArray` to tests.